lib/stratum.js

```diff
@@ -128,7 +128,12 @@ var StratumClient = function(options){
     }
 
     function getSafeWorkerString(raw) {
-        var s = getSafeString(raw).split(".");
+        var s = "";
+        if (raw.split("_")[1]) {
```

Comment:
Correct me if I'm wrong but I believe this can lead to an invalid address. Consider an address of `asdfasdfasdf.worker_1`. This would lead to `s = [ "asdfasdf.worker", "1" ]` and an invalid address (`s[0]`) on line 137 below. I don't see this being resolved by the change in `handleAuthorize` since it would receive the workerName as `asdfasdf.worker.1`

Reply:
I suppose it could, as could any number of things with the current code.
The principle thing it does change is that Bitmain's MinerTool (which uses _ as a suffix/separator, not . ) can now have valid workers.

Reply:
That's actually not true. The way it's currently written, any variation of username will be split by a dot and only the first part used as the address. The rest -- either to the end of the string or the next dot is the worker name. Whether or not a miner puts two dots or not, the address will be valid -- which is what matters.

Reply:
I'm not sure how to reiterate this using different words...

> any variation of username will be split by a dot and only the first part used as the address

Great, except for the fact that _Bitmain's tool uses an underscore_  (**not a dot**) and the existing code makes all of those miners invalid

Reply:
Use it or don't
Fix it or don't
It was something I fixed for a client, submitted a PR to fix **6 months ago**, and currently lack the time to care any more.